dapps/kiosk/src/routes/Home.tsx

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useWalletKit } from '@mysten/wallet-kit';
+import { useRpc } from '../hooks/useRpc';
+import { KIOSK_OWNER_CAP, createKioskAndShare } from '@mysten/kiosk';
+import { useEffect, useState } from 'react';
+import { TransactionBlock, getObjectFields, getObjectId } from '@mysten/sui.js';
+import { useTransactionExecution } from '../hooks/useTransactionExecution';
+import { KioskData } from '../components/Kiosk/KioskData';
+import { Loading } from '../components/Base/Loading';
+import { localStorageKeys } from '../utils/utils';
+import { SuiConnectButton } from '../components/Base/SuiConnectButton';
+
+function Home() {
+  const { currentAccount } = useWalletKit();
+  const [kioskIds, setKioskIds] = useState<string[]>([]);
+  const [loadingCap, setLoadingCap] = useState<boolean>(false);
+  const { signAndExecute } = useTransactionExecution();
+  const [kioskId, setKioskId] = useState<string | null>(null);
+  const provider = useRpc();
+
+  const findUserKiosks = async () => {
+    if (!currentAccount?.address) return;
+
+    setLoadingCap(true);
+
+    // get kiosk owner Cap objects
+    const kiosks = await provider.getOwnedObjects({
+      owner: currentAccount.address,
+      filter: { StructType: `${KIOSK_OWNER_CAP}` },
+      options: {
+        showContent: true,
+      },
+    });
+
+    const kioskIdList = kiosks?.data?.map((x) => getObjectFields(x)?.for);
+    setKioskIds(kioskIdList);
+
+    const kioskOwnerCaps = kiosks?.data.map((x) => getObjectId(x));
+
+    // save to localStorage for easy retrieval throughout the app.
+    localStorage.setItem(localStorageKeys.USER_KIOSK_ID, kioskIdList[0]);
```

Comment:
If you use tanstack query you could add a persister but honestly I'd just nuke this all and depend on the in-memory cache